redwood-layout-view/src/main/kotlin/app/cash/redwood/layout/view/ViewBox.kt

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (C) 2023 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.redwood.layout.view
+
+import android.content.Context
+import android.view.Gravity
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewGroup.LayoutParams.MATCH_PARENT
+import android.view.ViewGroup.LayoutParams.WRAP_CONTENT
+import android.view.ViewGroup.MarginLayoutParams
+import android.widget.FrameLayout
+import androidx.core.view.children
+import androidx.core.view.updateLayoutParams
+import app.cash.redwood.Modifier
+import app.cash.redwood.layout.api.Constraint
+import app.cash.redwood.layout.api.CrossAxisAlignment
+import app.cash.redwood.layout.modifier.Height
+import app.cash.redwood.layout.modifier.HorizontalAlignment
+import app.cash.redwood.layout.modifier.VerticalAlignment
+import app.cash.redwood.layout.modifier.Width
+import app.cash.redwood.layout.widget.Box
+import app.cash.redwood.ui.Density
+import app.cash.redwood.ui.Margin
+import app.cash.redwood.widget.ViewGroupChildren
+import kotlin.math.roundToInt
+
+internal class ViewBox(
+  context: Context,
+) : Box<View> {
+
+  private val density = Density(context.resources)
+
+  override var modifier: Modifier = Modifier
+
+  private var defaultHorizontalAlignment = CrossAxisAlignment.Start
+  private var defaultVerticalAlignment = CrossAxisAlignment.Start
+
+  override val value = object : FrameLayout(context) {
+    init {
+      layoutParams = LayoutParams(MATCH_PARENT, MATCH_PARENT)
+
+      // TODO: Remove this.
+      setBackgroundColor(0xFFFFFF66.toInt())
+    }
+
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+      val widgetChildren = this@ViewBox.children.widgets
+
+      for (child in widgetChildren) {
+        child.value.layoutParams = toLayoutParams(child.modifier)
+        child.value.requestLayout()
+      }
+
+      // Get width and height values from layout parameters
+      val width = when (layoutParams.width) {
+        LayoutParams.MATCH_PARENT -> widthMeasureSpec
+        LayoutParams.WRAP_CONTENT -> widgetChildren.maxOfOrNull { it.value.measuredWidth } ?: 0
+        else -> layoutParams.width
+      }
+
+      val height = when (layoutParams.height) {
+        LayoutParams.MATCH_PARENT -> heightMeasureSpec
+        LayoutParams.WRAP_CONTENT -> widgetChildren.maxOfOrNull { it.value.measuredHeight } ?: 0
```

Comment:
not sure if this is the best way to capture this, but leaving it up to `super.onMeasure` to figure this out didn't really work